StyleCop.Analyzers/StyleCop.Analyzers/Helpers/TriviaHelper.cs

```diff
@@ -95,5 +96,81 @@ internal static int IndexOfTrailingWhitespace(SyntaxTriviaList triviaList)
 
             return (whiteSpaceStartIndex < triviaList.Count) ? whiteSpaceStartIndex : -1;
         }
+
+        /// <summary>
+        /// Strips all trialing whitespace trivia from the trivia list until a non-whitespace trivia is encountered.
+        /// end-of-line trivia are ignored, but kept.
+        /// </summary>
+        /// <param name="triviaList">The trivia list to strip of its trailing whitespace.</param>
+        /// <returns>The modified triviaList.</returns>
+        internal static SyntaxTriviaList WithoutTrailingWhitespace(this SyntaxTriviaList triviaList)
```

Comment:
I have some concerns about potentially confusing this with `SpacingExtensions.WithoutTrailingWhitespace`, but I'm not yet sure what course of action I would recommend.